internal/core/command/controller/messaging/external.go

```diff
@@ -124,7 +126,7 @@ func commandRequestHandler(requestTimeout time.Duration, dic *di.Container) mqtt
 			return
 		}
 
-		externalResponseTopic := common.BuildTopic(externalMQTTInfo.Topics[common.ExternalCommandResponseTopicPrefixKey], deviceName, url.QueryEscape(commandName), method)
+		externalResponseTopic := common.BuildTopic(externalMQTTInfo.Topics[common.ExternalCommandResponseTopicPrefixKey], deviceName, commandName, method)
```

Comment:
Why are we no longer escaping the commandName? We are assuming it has already been escaped when sent via External MQTT, correct?

If this is the case then the commit/PR title should be a `fix` for Regex command via messaging

Reply:
The original implementation unescapes the commandName in L115 and escapes it back here.

During testing I realized `QueryEscape` function wil not escape `.`(dot), which is quite common in regex: https://go.dev/play/p/C20mt4CUbaG

This change handles this case and you're right this should be a `fix`